src/BuildIt.Forms/BuildIt.Forms.Controls/Platforms/Android/CameraPreviewControlRenderer.cs

```diff
@@ -379,24 +448,38 @@ internal void OpenCamera(int width, int height)
             CameraDevice = null;
             SetUpCameraOutputs(width, height);
             ConfigureTransform(width, height);
+            var errorOccurred = false;
             var manager = (CameraManager)Context.GetSystemService(Context.CameraService);
             try
             {
                 if (!CameraOpenCloseLock.TryAcquire(2500, TimeUnit.Milliseconds))
```

Comment:
Can we have some comments around why this is necessary - right now it feels like we're adding protection against certain scenarios but it's not clear to anyone reading the code what that scenario is